core/utils.py

```diff
@@ -172,4 +172,8 @@ def make_grid(
 
 def remove_degree_2_nodes(fua: int | str) -> geopandas.GeoDataFrame:
     """Remove [interstitial / non-articulation / degree 2] nodes from road network."""
-    return momepy.remove_false_nodes(read_original(fua))
+    return (
+        momepy.remove_false_nodes(read_original(fua))
+        .pipe(lambda df: df[~df.geometry.duplicated()].reset_index(drop=True))
+        .pipe(lambda df: momepy.remove_false_nodes(df))
+    )
```

Comment:
```suggestion
    df = momepy.remove_false_nodes(read_original(fua))
    df = df[~df.geometry.duplicated()].reset_index(drop=True)
    return momepy.remove_false_nodes(df)
```

Sorry, I hate pipes.